Fix(general-keyboard): Automatically switch to letter keys after command selection

smalik2811 opened the pull request:
### Contributor checklist

-   [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
-   [x] I have tested my code with the `./gradlew lintKotlin detekt test` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Android/blob/main/CONTRIBUTING.md#testing)

---

### Description

This commit addresses issue #298 by:

1. **Automatically switching the keyboard to 'keyboardLetters' after a Scribe command (e.g., translate, conjugate, plural) is selected.** This improves the user experience by streamlining the workflow and reducing the need for manual keyboard mode changes.

2. **Switching the keyboard to 'keyboardLetters' when the user switches to the Idle mode.** This ensures consistent keyboard behavior and prevents unexpected input modes.

This enhancement enhances the overall usability of Scribe by providing a more intuitive and efficient interaction flow for users.

Closes #298

### Related issue

-   #298

andrewtavis commented:
Minor feedback here, @smalik2811: I think that we don't need the keyboard switching back to the letter keys each time the state goes to idle. Wasn't quite as clear on this as I could have been :)

An example would be:

- The user is on symbols
- The user presses the Scribe key
- The user presses exit as they're exploring the keyboard for the first time or just decided not to use a command
- In this case the keyboard should still be symbols

What I meant was that each time after the user executes a command that we should switch back to the letter keys, but then that would be handled in the functions that execute the commands, not on a switch to letters whenever the keyboard switches to idle mode.

Do you want to send along the change such that this just switches the keyboard to letters when we enter a command mode, and then we can keep the letter keys switch in mind for when we work on the command functions in #266, #270 and #272?

Thanks for the help with this! 😊

CC @angrezichatterbox on this as well 👋

angrezichatterbox commented:
So does this mean that the keyboard should be on Keys on all modes except select command mode and idle mode. Like when the user click on a particular command  translate plural or conjugate he is switched to the keys. Am I right ?

andrewtavis commented:
Yes, exactly, @angrezichatterbox :) If they're on symbols and want to enter a word into a command, then the first thing they'll need to do is switch to letters, and they'll need to do it basically always, so let's do it for them 😊

smalik2811 commented:
Hey @andrewtavis, thanks for the feedback 😅. I totally missed the point that the user can return back to the Idle mode straight from the Command Select mode and this could result in the issue of unnecessaryly switching back from symbols to letters.

I have removed the part causing the issue. Now changes only includes the code to switch symbols to letters then the user actually enters the command mode.

The changes were affecting the capitalization of Noun in German, I have also fixed that.

andrewtavis commented:
So far so good, @smalik2811! Thanks for the quick reaction :) One final comment here is that the German noun capitalization is still being effected by this, or maybe it's broken and we need to fix it. When we do the plural command on the German keyboard the keyboard should switch to letter keys and it should also be capitalized for one key.

Do you want to try to send along a fix for this? 😊

smalik2811 commented:
Hey @andrewtavis, I tested the app on a virtual device. I first switched to the symbol keyboard from the idle mode and then entered the Plural mode. The keyboard switched to the letters keyboard and the letters are capitalized as expected.


[German capitalization.webm](https://github.com/user-attachments/assets/61ca51cf-c6c6-4000-b789-2e73aad73e28)


I must have missed something 😅, could you specify the issue you noticed with the German noun capitalization.

andrewtavis reviewed (approved):
Sorry for the confusion, @smalik2811! Thanks much for the video showing that all's working. I just updated to Android Studio Ladybug and I think that there was just some issues where the changes weren't being reflected. Needed to leave the keyboard and come back to it and it's all great now 😊

Thanks for the continued support!

smalik2811 commented:
No problem at all! Glad you got it sorted out and we have resolved another issue 😊
